pkg/cloud/services/network/vpc.go

```diff
@@ -211,10 +241,25 @@ func (s *Service) createVPC() (*infrav1.VPCSpec, error) {
 		input.AmazonProvidedIpv6CidrBlock = aws.Bool(s.scope.VPC().IsIPv6Enabled())
 	}
 
-	if s.scope.VPC().CidrBlock == "" {
-		s.scope.VPC().CidrBlock = defaultVPCCidr
+	if s.scope.VPC().IPAMPool != nil {
+		ipamPoolID, err := s.getIPAMPoolID()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get IPAM Pool ID")
+		}
+
+		if s.scope.VPC().IPAMPool.NetmaskLength == 0 {
```

Comment:
I'm wondering if this would cause any problems for IPv6 based IPAMs?

Reply:
Hey @Skarlso!

If my understanding is correct, all settings in `spec.network.vpc` applies only to `ipv4`. For `ipv6` there is a separate configuration in `spec.network.vpc.ipv6`. 

And according to webhook, ipv6 cannot be used with managed clusters:
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/29c442b8a72ddb5677047cf6fcb80358aad8c440/api/v1beta2/awscluster_webhook.go#L182-L193

As for EKS, there is no `ipv6` configuration in `AWSManagedClusterSpec`.

Reply:
That's not completely true. The IPv6 part there only tells ipv6 specific settings. But this is IPAM. I wouldn't want to copy and replicate every setting there is to the ipv6 field. In the code, we check if ipv6 is enabled via that setting. But other than that some things are set separately by dedicated fields.

Reply:
> I wouldn't want to copy and replicate every setting there is to the ipv6 field.
For IPv6 there should be a separate IPAM pool. So if a user wants to use IPAM for dual-stack vpc, he must create two different pools, and in this case IPAM parameter should be duplicated :(

Reply:
Oh, that's interesting. I haven't considered that. 🤔 That messes things up a bit indeed. :D

Reply:
I guess it's fine then.